File: Test/shop-parent/shop-order-api/src/main/java/com/gec/shop/order/pojo/Order.java
package com.gec.shop.order.pojo;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.Version;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;

/**
 * 订单实体类
 */
@Data
@TableName("t_order")
public class Order implements Serializable {
    @TableId(type = IdType.AUTO)
    private Long id;
    @TableField("order_no")
    private String orderNo;
    private Long uid;
    private String username;
    private Long pid;
    private String productName;
    private BigDecimal productPrice;
    private Integer number;
    private String status;
    @Version
    private Integer version;
    private java.time.LocalDateTime createTime;
    private java.time.LocalDateTime updateTime;
}

File: shop-web/src/api/types/order.ts
// 订单相关类型定义

export interface Order {
  id: number
  pid: number
  uid: number
  username: string
  productName: string
  productPrice: number
  number: number
  status: string
  version: number
  createTime?: string
  updateTime?: string
}

export interface OrderCreateRequest {
  pid: number
  uid: number
}

export interface OrderCreateResponse {
  code: number
  msg: string
  orderId: number
}

export interface OrderStatusUpdateRequest {
  orderId: number
  status: string
}
